### ISSUE TITLE
Release cycle on NuGet should be aligned with VS releases

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#69478]
At the time of writing this issue, the latest on NuGet is `4.7.0-2.final`, while the current VS preview is `17.8 Preview 1`.

So Roslyn packages on NuGet are **very much** behind VS versioning. The current Roslyn version on NuGet corresponds to 17.7 Preview 2.

And not only it's behind by preview versions only, but also that VS 17.7 is now stable and the 4.7 on NuGet is still preview (and very old preview).

**Comment by AraHaan** [comment]
I agree, I feel that VS should actually be using the nuget version instead to make it easier on deployment for both VS and people needing to use roslyn outside of VS via nuget anyways.

**Comment by genlu** [comment]
Releasing NuGet packages alone side VS releases is our goal, but sometimes it gets delayed. For example, we might need to wait for all our dependencies to be published on nuget.org first, some of those are owned by other teams and have to be published manually.

FYI @ryzngard for 17.8p1 packages specifically

**Comment by Youssef1313** [comment]
Yup thanks @genlu. Already talked with @ryzngard on Discord and knew that such delays can sometimes happen. Sometimes the time gap is quite large though. If something in the process can change to reduce such delays (e.g., reduce manual work), that would be great.

**Comment by ryzngard** [comment]
Speaking on delays on this, we have had two problems: 

`17.7-final` packages were delayed because a dependency that wasn't published to nuget. We had to work with the team before we could do a manual verification of the packages, do another release pipeline, and then finally release.

`17.8p1` is delayed while we determine what appears to be a nuget authoring issue of an upstream dependency that would cause errors if the packages were used side by side. We're still working on resolving this one.

Some inside look into the process for us: we do a rotation of infrastructure duties on a monthly, and do our best to publish in a timely manner. Unfortunately this is a very manual process still to find issues and resolve them. I appreciate issues like this since it represents a real pain point for users. I'm not sure what we can do about the root cause of these delays, since the failure step has been manual validation we get from a 3rd party to ensure correctness within the full ecosystem, but I'll think on it.
